builtin/mainmenu/tab_online.lua

```diff
@@ -363,12 +363,18 @@ local function matches_query(server, query)
 	return name_matches and 50 or description_matches and 0
 end
 
+local pre_search_selection = nil
+
 local function search_server_list(input)
 	menudata.search_result = nil
 	if #serverlistmgr.servers < 2 then
 		return
 	end
 
+	if not pre_search_selection then
```

Comment:
You can replace this `if` with `pre_search_selection = pre_search_selection or find_selected_server()`